target/ppc/translate/storage-ctrl-impl.c.inc

```diff
@@ -40,26 +40,30 @@ static bool do_tlbie(DisasContext *ctx, arg_X_tlbie *a, bool local)
     }
 
     if (ctx->pr) {
-        /* tlbie is privileged... */
+        /* tlbie[l] is privileged... */
         GEN_PRIV2(true);
     } else if (!ctx->hv) {
-        if (!ctx->gtse || (!prs && ctx->hr)) {
+        if ((!prs && ctx->hr) || (!local && !ctx->gtse)) {
             /*
-             * ... except when GTSE=0 or when PRS=0 and HR=1, making it
-             * hypervisor privileged.
+             * ... except when PRS=0 and HR=1, or when GTSE=0 for tlbie,
+             * making it hypervisor privileged.
              */
             GEN_PRIV2(true);
         }
     }
 
-    if (NARROW_MODE(ctx)) {
+    if (!local && NARROW_MODE(ctx)) {
         TCGv t0 = tcg_temp_new();
         tcg_gen_ext32u_tl(t0, cpu_gpr[rb]);
         gen_helper_tlbie(cpu_env, t0);
         tcg_temp_free(t0);
     } else {
         gen_helper_tlbie(cpu_env, cpu_gpr[rb]);
     }
+
+    if (local)
+        return true;
```

Comment:
QEMU style requires braces in all indented statements, even if it's a single line.

Reply:
Right, sometimes I forget about this...

Reply:
Fixed.